src/cr/cube/distributions/wishart.py

```diff
@@ -0,0 +1,135 @@
+# encoding: utf-8
+
+"""The CDF of the largest eigenvalue of a Wishart distribution.
+
+Compute the CDF of the Wishart distribution using the algorithm of
+Chiani (2014). The Wishart CDF is not defined in scipy.stats.wishart, nor in
+tensorflow_probability.distributions.wishart. At the time of writing, the only
+implementation of this CDF is the R package rootWishart. This largely
+follows that, but using scipy implementations rather than Boost, and omitting
+multi-precision support.
+"""
+
+from __future__ import division
+
+import numpy as np
+from scipy.special import gamma, gammainc
+from scipy.linalg import det
+from scipy.constants import pi
+
+try:
+    xrange
+except NameError:
+    xrange = range
+
+
+def wishartCDF(X, n_min, n_max):
+    w_p = np.zeros(X.shape)
+    for i, x in np.ndenumerate(X):
+        w_p[i] = _wishart_pfaffian(x, n_min, n_max)
+    return _normalizing_const(n_min, n_max) * w_p
+
+
+def _wishart_pfaffian(x, n_min, n_max):
+    """Value providing the CDF of the largest eigenvalue of a Wishart(n, p)
+    evaluated at x.
+
+    if (n_min != n_mat) {
+        // Fill in extra column
+        VectorXmp alpha_vec(n_min);
+        T alpha_last = 0.5*(n_max + n_min + 1);
+        for (int i = 0; i < n_min; i++) {
+            alpha_vec(i) = alpha + i + 1;
+            A(i, n_min) = pow(2, -alpha_last)*boost::math::gamma_p(alpha_vec(i), 0.5*xx)/boost::math::tgamma(alpha_last);
+            A(n_min, i) = - A(i, n_min);
+        }
+    }
+    """
+    size = n_min + (n_min % 2)
+    alpha = 0.5 * (n_max - n_min - 1)
+    A = np.zeros([size, size])
+    alpha_ind = np.arange(n_min, dtype=np.float)
+    alpha_vec = alpha_ind + alpha + 1
+    gammainc_a = gammainc(alpha_vec, 0.5 * x)
+    if n_min != size:
+        alpha_ind = alpha_ind.astype(np.int)
+        other_ind = np.full(n_min, size - 1, dtype=np.int)
+        alpha_last = (n_max + n_min + 1) / 2
+        A[other_ind, alpha_ind] = (
+            np.float_power(2, -alpha_last) * gammainc_a / gamma(alpha_last)
+        )
+        # TODO check that this is not transposed
+        A[alpha_ind, other_ind] = -A[other_ind, alpha_ind]
+    """
+
+    for (int i = 0; i < n_min; i++) {
+        p(i) = boost::math::gamma_p(alpha + i + 1, 0.5*xx);
+        g(i) = boost::math::tgamma(alpha + i + 1);
+    }
+    for (int i = 0; i < (2*n_min - 2); i++) {
+        q(i) = pow(0.5, 2*alpha + i + 2) * boost::math::tgamma(2*alpha+i+2)*boost::math::gamma_p(2*alpha + i + 2, xx);
+    }
+    """
+    p = gammainc_a
+    g = gamma(alpha_vec)
+    q_ind = np.arange(2 * n_min - 2)
+    q = (
+        np.float_power(0.5, 2 * alpha + q_ind + 2)
```

Comment:
could use just `power` here but I figured explicit>implicit for cases where the base is int.

Reply:
```suggestion
        q_vec = 2 * alpha + q_ind + 2
        np.float_power(0.5, q_vec)
```